src/com/amplitude/api/DefaultNetworkClient.java

```diff
@@ -0,0 +1,33 @@
+package com.amplitude.api;
+
+import okhttp3.FormBody;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+
+import java.io.IOException;
+
+public class DefaultNetworkClient implements NetworkClient {
+
+    @Override
+    public Response uploadEvents(EventUploadRequest eventUploadRequest, OkHttpClient client) throws IOException, IllegalArgumentException {
```

Comment:
@danmalafaia I notice in your API choice that you are still returning an Okhttp3 response object. In your custom client, are you also going to be using the Okhttp3 library? 

I think it might be cool to let the network client to decide what networking protocol/library to use, although I don't feel strongly either way as long as you're okay using the Okhttp3 api as well. In theory the parent just needs the string response body and int response code from whatever response to do the next part.

Reply:
We can create a custom response object not related to Okhttp3 library with only the response code and response body.

However we can't remove the reference from Okhttp3 from the parameters without a big refactor. As the library is already using it by default and in the ssl pinned version it's simpler to keep it.